okhttp/src/main/java/okhttp3/Headers.java

```diff
@@ -334,6 +339,17 @@ public Builder addAll(Headers headers) {
       return this;
     }
 
+    /**
+     * Add a Date header with the current time based on
+     * https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date .
+     */
+    public Builder withDate() {
```

Comment:
Thoughts on a different signature that makes the caller call `new Date()` ?

```
public Builder add(String name, Date value)
public Builder set(String name, Date value)
```

Reply:
... I think it’d be simpler to give callers a `DateInterceptor` than a `withDate()` method ’cause then they only need to configure it once.

Reply:
With regards to the add/set methods I think that would be fine, I was just trying to implement it as I understood it from the original feature request.

Agreed on the DateInterceptor as a good approach. This is typically how I would do it in my clients.

I'm happy to do it either or both ways.

Reply:
Okay I think my preference is add/set methods that take a Date, and a sample interceptor in our Recipes or Docs somewhere.

I don’t want to provide a built-in Interceptor for this ’cause I want people to get their hands dirty and write their own interceptors. I think it’s actually the nicer way to use the library.

Reply:
Sounds good, I will update the diff tomorrow with those changes.

Reply:
Could we also support the Java 8 Data/Time API? n.b. I haven't considered the correct types to map, and behaviour for timezones etc.

See https://github.com/square/okhttp/pull/4272/files for the required annotations.

Reply:
@yschimke do you need the Java 8 support in this diff or could it be a separate one?

Reply:
@swankjesse Made those changes and add a recipe.

Reply:
@jjshanks For me, I think just add it as part of this PR.  I really don't want to be using Date() in my code :)